experimental/c-cpp/manager.py

```diff
@@ -772,6 +852,96 @@ def evaluate_heuristic(test_dir, result_to_validate, fuzzer_intrinsics,
   shutil.copytree(fuzzer_gen_dir,
                   os.path.join(outdir, os.path.basename(fuzzer_gen_dir)))
 
+  # Create an OSS-Fuzz integration and ClusterFuzzLite integration
+  create_clean_oss_fuzz_from_success(
+      github_repo, os.path.join(outdir, os.path.basename(fuzzer_gen_dir)))
+  create_clean_clusterfuzz_lite_from_success(
+      github_repo, os.path.join(outdir, os.path.basename(fuzzer_gen_dir)))
+
+
+def create_clean_oss_fuzz_from_success(github_repo, success_dir):
+  """Converts a successful out dir into a working OSS-Fuzz project."""
+  oss_fuzz_folder = os.path.join(success_dir, 'oss-fuzz-project')
+  os.makedirs(oss_fuzz_folder)
+
+  # Project yaml
+  project_yaml = {
+      'homepage': github_repo,
+      'language': 'c++',
+      'primary_contact': 'add_your_email@here.com',
+      'main_repo': github_repo
+  }
+  with open(os.path.join(oss_fuzz_folder, 'project.yaml'), 'w') as project_out:
+    yaml.dump(project_yaml, project_out)
+
+  # Copy fuzzer
+  shutil.copy(os.path.join(success_dir, 'empty-fuzzer.cpp'),
+              os.path.join(oss_fuzz_folder, 'fuzzer.cpp'))
+
+  # Create Dockerfile
+  project_repo_dir = github_repo.split('/')[-1]
+  dockerfile = CLEAN_DOCKER % (github_repo, project_repo_dir, project_repo_dir)
+  with open(os.path.join(oss_fuzz_folder, 'Dockerfile'), 'w') as docker_out:
+    docker_out.write(dockerfile)
+
+  # Build file
+  with open(os.path.join(success_dir, 'build.sh'), 'r') as f:
+    build_content = f.read()
+  split_build_content = build_content.split("\n")
+
+  original_build_folder = split_build_content[1].split('/')[-1]
+
+  clean_build_content_lines = split_build_content[:1] + split_build_content[4:]
+  clean_build_content = '\n'.join(clean_build_content_lines).replace(
+      '/src/generated-fuzzer',
+      '$OUT/fuzzer').replace('empty-fuzzer', 'fuzzer').replace(
+          '/src/', '$SRC/').replace(original_build_folder, project_repo_dir)
```

Comment:
A silly question:
What is the goal of modifying the `build.sh` here?
In particular, why can we assume the line numbers (`1`, `4`) in this line?
```python
clean_build_content_lines = split_build_content[:1] + split_build_content[4:]
```

Reply:
The goal is to convert it from how it looks during auto-gen testing/evaluation stages to be something that can work proper with a clean OSS-Fuzz project.

Build used during testing inside of base-builder container:

```sh
#!/bin/bash
rm -rf //src/test-fuzz-build-7
cp -rf /src/project_source /src/test-fuzz-build-7
cd /src/test-fuzz-build-7
mkdir fuzz-build
cd fuzz-build
cmake -DCMAKE_VERBOSE_MAKEFILE=ON ../
make V=1 || true
```

Build for clean OSS-Fuzz project

```sh
#!/bin/bash
mkdir fuzz-build
cd fuzz-build
cmake -DCMAKE_VERBOSE_MAKEFILE=ON ../
make V=1 || true
```

The lines you highlight cut out the following from the build script:

```sh
rm -rf //src/test-fuzz-build-7
cp -rf /src/project_source /src/test-fuzz-build-7
cd /src/test-fuzz-build-7
```